File: client/src/app/modules/reporting/services/incident.type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { IncidentType } from '../../reporting/model/IncidentType';

@Injectable({
  providedIn: 'root'
})
export class IncidentTypeService extends AppRestService<IncidentType>{

  constructor(http: HttpClient) {
    super(http, "reporting");
  }

  getIncidentTypes() {
    return this.getList(new HttpParams(), "incident-types");
  }
  
   addOrUpdateIncidentType(incidentType: IncidentType) {
    return this.addOrUpdate(new HttpParams(), "incident-type", incidentType, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/osiris/reporting/service/IncidentTypeService.java
package com.jss.osiris.modules.osiris.reporting.service;

import java.util.List;

import com.jss.osiris.modules.osiris.reporting.model.IncidentType;

public interface IncidentTypeService {
    public List<IncidentType> getIncidentTypes();

    public IncidentType getIncidentType(Integer id);

    public IncidentType addOrUpdateIncidentType(IncidentType incidentType);
}
